.github/copilot-instructions.md

```diff
@@ -0,0 +1,298 @@
+
+- **Size**: Large monorepo (~1M+ lines of code)
+- **Backend**: Ruby 3.4.5, Rails ~8.0.3
+- **Frontend**: Node.js 22.15.0, npm 10.1.0+, Angular 20+, TypeScript
+- **Database**: PostgreSQL (required)
+- **Architecture**: Hybrid Ruby on Rails + Angular SPA with Stimulus/Turbo
+
+## Critical Setup Requirements
+
+### Ruby and Node Versions
+**ALWAYS verify versions before building:**
+- Ruby: `3.4.5` (see `.ruby-version`)
+- Node: `^22.15.0` (see `package.json` engines)
+- Bundler: Latest 2.x
+
+### Development Environment Options
+
+**Docker (Recommended for Quick Start)**
+```bash
+# ALWAYS run these commands in sequence:
+cp .env.example .env
+cp docker-compose.override.example.yml docker-compose.override.yml
+docker compose run --rm backend setup
+docker compose run --rm frontend npm install
+docker compose up -d backend
+# Access at http://localhost:3000
+```
+
+**Local Development Setup**
+```bash
+# Install dependencies (ALWAYS run in this order):
+bundle install                    # Install Ruby gems
+cd frontend && npm ci && cd ..   # Install Node packages (use 'ci' not 'install' for reproducibility)
+bundle exec rake db:migrate      # Setup database
+bundle exec rails openproject:plugins:register_frontend assets:export_locales
+
+# Start services (use bin/dev for all-in-one):
+bin/dev                          # Starts Rails, Angular, and Good Job worker
+# OR manually:
+# Terminal 1: bundle exec rails server
+# Terminal 2: npm run serve
+# Terminal 3: bundle exec good_job start
+```
+
+**Important**: The `config/database.yml` file MUST NOT exist when using Docker. Delete or rename it if present.
+
+## Building and Testing
+
+### Linting (Run Before Committing)
+
+**Ruby (Rubocop)**
+```bash
+bundle exec rubocop                          # Check all files
+bin/dirty-rubocop --uncommitted              # Check only uncommitted changes
+bin/dirty-rubocop --uncommitted --force-exclusion {files}  # Check specific files
+```
+
+**JavaScript/TypeScript (ESLint)**
+```bash
+cd frontend
+npx eslint src/                              # Lint all frontend code
+cd ..
+```
+
+**ERB Templates (erb_lint)**
+```bash
+erb_lint {files}                             # Lint ERB template files
+```
+
+**Install Git Hooks** (optional but recommended):
+```bash
+bundle exec lefthook install                 # Sets up pre-commit hooks for linting
+```
+
+### Running Tests
+
+**Backend Tests (RSpec)**
+```bash
+# Run specific tests (ALWAYS preferred over running all tests):
+bundle exec rspec spec/models/user_spec.rb              # Single file
+bundle exec rspec spec/models/user_spec.rb:42           # Single line
+bundle exec rspec spec/features                         # Directory
+
+# Run all tests (slow, ~40 minutes on CI):
+bundle exec rspec
+
+# Parallel execution (faster):
+bundle exec rake parallel:spec
+
+# With Docker:
+docker compose run --rm backend-test bundle exec rspec spec/features/work_package_show_spec.rb
```

Comment:
IIRC running specs requires additional quoting (that might be a bug, but it's what it is):

```suggestion
docker compose run --rm backend-test "bundle exec rspec spec/features/work_package_show_spec.rb"
```